demos/tests/cases.py

```diff
@@ -23,31 +23,47 @@
 TestCase = collections.namedtuple('TestCase', ['options'])
 
 class NativeDemo:
-    def __init__(self, name, test_cases):
+    def __init__(self, name, test_cases, subdirectory=None):
```

Comment:
I think instead of specifying the subdirectory and name as two separate parameters, you should remove the `name` parameter and derive it as `subdirectory.replace('/', '_')`. The multichannel demos' executable names will have to be updated to match this convention, but it's been long overdue anyway.

Reply:
updated